### PR TITLE
get rid of dependencies fs-extra and tempy

### PR DESCRIPTION
**Pull request by dotlambda**
Why use dependencies that you'll have to trust if the functionality is built into Node.js?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,18 +36,16 @@
...
     "lodash.pick": "^4.4.0",
     "napi-macros": "^2.0.0",
     "node-fetch": "^2.6.0",
     "node-gyp-build": "^4.1.0",
     "split2": "^3.1.1",
-    "tempy": "^0.3.0",
     "typescript": "^3.7.5"
   },
   "devDependencies": {
+    "@types/node": "^16.4.1",
```
Comment (dotlambda) [inline package.json:48]: I had to bump the version because `fs/promise` is only available from Node.js 14 on. Is that a problem?

Comment (dotlambda) [reply package.json:48]: Should I use
```suggestion
    "@types/node": "^14.0.0",
```
?

Comment (Jikstra) [reply package.json:48]: Probably. Probably not. We definetly should make sure that we bump the nodejs versions on all cis scripts too. And see if everything still works :)

Comment (dotlambda) [reply package.json:48]: > make sure that we bump the nodejs versions on all cis scripts too

Done, except for `.github/workflows/build-and-test-integration-linux.yml` where I don't understand how Node.js is installed.

Also, does it make sense to still run CI on Ubuntu 16.04?

**Pull request merged**